main.tf

```diff
@@ -62,7 +62,17 @@ resource "aws_dynamodb_table" "basic-dynamodb-table" {
       range_key = try(index.value.range_key, null)
       read_capacity =  var.billing_mode == "PROVISIONED" ? var.read_capacity :  null
       write_capacity =  var.billing_mode == "PROVISIONED" ? var.write_capacity :  null
-      projection_type    = "ALL"
+      projection_type    = try(index.value.projection_type, "ALL")
+      non_key_attributes = try(index.value.projection_type, "ALL") == "INCLUDE" ? try(index.value.non_key_attributes, null) : null
+    }
+  }
+  
+  dynamic "ttl" {
+    for_each = var.ttl_attribute_name != null ? [1] : []
```

Comment:
Hi @evgenygigi , the ttl_attribute_name is string and not a list, and I am confused with the foreach loop that you are having here. Can you please explain?

Reply:
Because of how dynamic block works https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks
If var.ttl_attribute_name is not null _(meaning it has a value)_, the for_each expression evaluates to [1], creating a single iteration of the block. 
If var.ttl_attribute_name is null, the expression evaluates to an empty list ([]), effectively skipping the block's execution.